### PR TITLE
Add documentation for Network Pivoting

### PR DESCRIPTION
**Pull request by cdelafuente-r7**
This adds documentation for `Network Pivoting` under the `Pentesting` section.

### CODE REVIEW

File: docs/metasploit-framework.wiki/Metasploit-Guide-Network-Pivoting.md [hunk 1]
```diff
@@ -0,0 +1,326 @@
+# Network Pivoting
```
Comment (adfoster-r7) [inline docs/metasploit-framework.wiki/Metasploit-Guide-Network-Pivoting.md:1]: What's the main difference between this and https://github.com/rapid7/metasploit-framework/blob/37fe3b909a298e46db21380ebcda7965ad6fe492/docs/metasploit-framework.wiki/Pivoting-in-Metasploit.md ? 👀

Comment (cdelafuente-r7) [reply docs/metasploit-framework.wiki/Metasploit-Guide-Network-Pivoting.md:1]: I didn't know there was some existing documentation about this. I just followed the internal "Docs ideas" document you wrote earlier this year, which is suggesting a "Network Pivoting" documentation under the Pentesting section. I believe something more focus on specific pentesting workflows: initial compromise, internal network reconnaissance, lateral movement, etc.

**Comment by gwillcox-r7** [comment]
Marking this as Delayed until we make a decision on above point and get that resolved since it will likely require merging this documentation somehow.

**Comment by gwillcox-r7** [comment]
Hi folks, is there any way I can help move this forwards? This has been sitting in the queue for a while now.

**Comment by cdelafuente-r7** [comment]
I'm closing this since it will be included in the existing Pivoting-in-Metasploit.md documentation in a separate PR.

**Pull request closed without merging**